sec2_the-ruby-programming-language/2.11_oop_intro/4_attr_setter_getter.rb

```diff
@@ -0,0 +1,36 @@
+# attr_, setter and getter
+
+class Student
+  # attr_accessor
+  # automatically creater getter and setter
+  attr_accessor :first_name, :last_name, :email
+
+  # attr_reader
+  # only allow getter (NO setter)
+  attr_reader :username
+  
+  @first_name
+  @last_name
+  @email
+  @username = "michelle1"
+  @password
+
+  def to_s
+    "First name: #{@first_name}"
+    "Username: #{username}"
+  end
+
+end
+
+michelle = Student.new
+michelle.first_name = "Michelle"
+michelle.last_name = "Loh"
+michelle.email = "michelle@example.com"
+
+puts michelle.first_name    # Michelle
+puts michelle.last_name     # Loh
+puts michelle.email         # michelle@example.com
+puts michelle.username      #                           //nothing is output
+
+# Why nothing is output in Line 33?
+# because we did not initialise the object but we initailise in the class
```

Comment:
Refer Line 36, 15
When do we need to initialise in a class and why do we do that?

Reply:
wat do you mean by `initialise in a class`?

Reply:
like this `@username = "michelle1"`